share/wake/lib/core/list.wake

```diff
@@ -226,3 +226,16 @@ export def unzip = match _
   Pair a b, t =
     def Pair u v = unzip t
     Pair (a, u) (b, v)
+
+# Group a list by some comparison function.
+# Elements which compare equal are placed into the same bucket.
+# Buckets preserve the order of the original list.
+# The buckets appear in the sort-order specified by cmp.
+# Example: seq 5 | groupBy (_/3 < /_3)
+#          (0,1,2,Nil), (3,4,Nil), Nil
```

Comment:
What does `(_/3 < /_3)` actually mean? 

`(something divided by 3) is less than (something divided by 3)` is how it reads to me, but that doesn't connect to what the function outputs, which is 2 lists

Reply:
Your reading is correct.
So it says group everything together that is equal after being divided by 3.

In other words, group (0,1,2), (3,4,5), (6,7,8), ...

Reply:
`/_3` is a typo.

Reply:
I've made a separate PR for this function; please review it independently.
This PR is just based on top of that one so I can test CI.

Reply:
my attempt at reading doesn't lead to understanding though

What does each symbol do?
What is `/_` or is it that `/_` is an operator, or that `_3` is the name of a function